libraries/abstractions/common_io/test/test_iot_pwm.c

```diff
@@ -417,42 +397,45 @@ TEST( TEST_IOT_PWM, AFQP_IotPwmAccuracy )
     IotGpioInterrupt_t xGpioInterruptB = eGpioInterruptRising;
 
     /* set up port B for interrupt */
-    xtestIotGpioHandleB = iot_gpio_open( ultestIotPwmGpioInputPin );
+    xtestIotGpioHandleB = iot_gpio_open( ulGpioInstance );
     TEST_ASSERT_NOT_EQUAL( NULL, xtestIotGpioHandleB );
 
-    lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioDirection, &xGpioDirectionB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
+    if( TEST_PROTECT() )
+    {
+        lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioDirection, &xGpioDirectionB );
+        TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
 
-    lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioPull, &xGpioPullB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
+        lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioPull, &xGpioPullB );
+        TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
 
-    lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioInterrupt, &xGpioInterruptB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
+        lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioInterrupt, &xGpioInterruptB );
+        TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
 
-    iot_gpio_set_callback( xtestIotGpioHandleB, prvPwmGpioInterruptCallback, &xGpioInterruptB );
+        iot_gpio_set_callback( xtestIotGpioHandleB, prvPwmGpioInterruptCallback, &xGpioInterruptB );
 
-    /* Initialize our counter to 0 */
-    ultestIotPwmIrqCounter = 0;
+        /* Initialize our counter to 0 */
+        ultestIotPwmIrqCounter = 0;
 
-    /* Open PWM handle */
-    xPwmHandle = iot_pwm_open( ultestIotPwmInstance );
-    TEST_ASSERT_NOT_EQUAL( NULL, xPwmHandle );
+        /* Open PWM handle */
+        xPwmHandle = iot_pwm_open( ulPwmInstance );
+        TEST_ASSERT_NOT_EQUAL( NULL, xPwmHandle );
 
-    /* Settup the pwm configuration */
-    xSetPwmConfig.ulPwmFrequency = ultestIotPwmFrequency;
-    xSetPwmConfig.ucPwmDutyCycle = ultestIotPwmDutyCycle;
-    xSetPwmConfig.ucPwmChannel = ultestIotPwmChannel;
+        /* Settup the pwm configuration */
+        xSetPwmConfig.ulPwmFrequency = ultestIotPwmFrequency;
+        xSetPwmConfig.ucPwmDutyCycle = ultestIotPwmDutyCycle;
+        xSetPwmConfig.ucPwmChannel = ulPwmChannel;
 
-    /* Set the pwm configuration */
-    lRetVal = iot_pwm_set_config( xPwmHandle, xSetPwmConfig );
-    TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
+        /* Set the pwm configuration */
+        lRetVal = iot_pwm_set_config( xPwmHandle, xSetPwmConfig );
+        TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
 
-    /* Start the PWM signal */
-    lRetVal = iot_pwm_start( xPwmHandle );
-    TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
+        /* Start the PWM signal */
+        lRetVal = iot_pwm_start( xPwmHandle );
+        TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
 
-    /* Delay for 5 sec. */
-    vTaskDelay( PWM_TEST_DURATION_MSEC / portTICK_PERIOD_MS );
+        /* Delay for 5 sec. */
```

Comment:
Note that this comment will no longer be accurate if someone changes the value of `PWM_TEST_DURATION_MSEC`.  There are several comments with this issue, e.g. line 455.